lib/records/RecUtils.cc

```diff
@@ -419,7 +419,7 @@ RecDataSetFromString(RecDataT data_type, RecData *data_dst, const char *data_str
     data_src.rec_float = atof(data_string);
     break;
   case RECD_STRING:
-    if (data_string && strcmp((data_string), "nullptr") == 0) {
+    if (data_string && (strlen(data_string) == 4) && strncmp((data_string), "NULL", 4) == 0) {
```

Comment:
Because amc wanted me to use strncmp(), so if you don't check the length, then it'd also match  any string starting with NULL.